File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c in ['(', '[', '{']:
                stack.append(c)
            else:
                if len(stack) == 0:
                    return False
                if stack[-1] == '(' and c == ')':
                    stack.pop()
                    continue
                if stack[-1] == '[' and c == ']':
                    stack.pop()
                    continue
                if stack[-1] == '{' and c == '}':
                    stack.pop()
                    continue
                return False 
        if len(stack) == 0:
            return True
        else:
            return False
                

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (int i = 0; i < s.length(); i++) {
            if (s[i] == ')' || s[i] == '}' || s[i] == ']') {
                if (st.empty()) return false;
                char curr = st.top();
                if (curr == '(' && s[i] == ')') st.pop();
                else if (curr == '{' && s[i] == '}') st.pop();
                else if (curr == '[' && s[i] == ']') st.pop();
                else return false;
            } else {
                st.push(s[i]);
            }
        }
        if (st.empty()) return true;
        else return false;
    }
};